packages/ui/src/components/List/ListContext.tsx

```diff
@@ -41,17 +42,27 @@ type ListProviderProps = {
   autoCollapse: boolean
   selectable: boolean
   expandButton: boolean
+  onSelectedChange?: Dispatch<SetStateAction<string[]>>
 }
 
 export const ListProvider = ({
   children,
   autoCollapse,
   selectable,
   expandButton,
+  onSelectedChange,
 }: ListProviderProps) => {
   const [expandedRowIds, setExpandedRowIds] = useState<RowState>({})
   const [selectedRowIds, setSelectedRowIds] = useState<RowState>({})
 
+  useEffect(() => {
+    if (onSelectedChange) {
+      onSelectedChange(
+        Object.keys(selectedRowIds).filter(row => selectedRowIds[row]),
+      )
+    }
+  }, [onSelectedChange, selectedRowIds])
```

Comment:
The idea is to avoir needing to make that watcher function as we use it multiple times and instead include it directly into the component. This was requested by the front team and I think it can be better to have a simplified version of this. Wdyt?

Reply:
We can avoid useEffect by placing onSelectedChange in selectAll, unselectAll, selectRow, unselectRow ? :thinking:

Reply:
as you wish, hard to judge value of this since I never faced the need of it but it can make sense I guess.

Except the fact that we could avoid the use avoid, i'm fine with this MR, I approve it